commit 0701214cd6e66585a999b132eb72ae0489beb724
Author: Chuck Lever <chuck.lever@oracle.com>
Date:   Fri Jul 22 15:08:04 2022 -0400

    SUNRPC: Fail faster on bad verifier
    
    A bad verifier is not a garbage argument, it's an authentication
    failure. Retrying it doesn't make the problem go away, and delays
    upper layer recovery steps.
    
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
    Reviewed-by: Jeff Layton <jlayton@kernel.org>
    Signed-off-by: Trond Myklebust <trond.myklebust@hammerspace.com>

diff --git a/net/sunrpc/clnt.c b/net/sunrpc/clnt.c
index b6781ada3aa8..a97d4e06cae3 100644
--- a/net/sunrpc/clnt.c
+++ b/net/sunrpc/clnt.c
@@ -2650,7 +2650,7 @@ rpc_decode_header(struct rpc_task *task, struct xdr_stream *xdr)
 
 out_verifier:
 	trace_rpc_bad_verifier(task);
-	goto out_garbage;
+	goto out_err;
 
 out_msg_denied:
 	error = -EACCES;